cmd/root/collection/cluster.go

```diff
@@ -65,19 +70,26 @@ func GetClusterLogs(namespace string, cs CopyStrategy, ddfs helpers.Filesystem,
 
 	// Loop over dremio pods
 	for _, pod := range pods {
-		kubectlArgs := []string{k, "-n", namespace, "get", "pods", string(pod), "-o", `jsonpath={.spec['containers','initContainers'][*].name}`}
-		containers, err := clusterExecutePod(kubectlArgs)
-		if err != nil {
-			simplelog.Errorf("trying to list containers from pod %v with error %v", pod, err)
-			return err
-		}
-		// Loop over each container, construct a path and log file name
-		// write the output of the kubectl logs command to a file
-		for _, container := range strings.Split(containers, " ") {
-			copyContainerLog(cs, ddfs, k, container, namespace, path, pod)
-		}
+		wg.Add(1)
+		go func(podname string) {
+			defer wg.Done()
+			kubectlArgs := []string{k, "-n", namespace, "get", "pods", string(podname), "-o", `jsonpath={.spec['containers','initContainers'][*].name}`}
+			containers, err := clusterExecutePod(kubectlArgs)
+			if err != nil {
+				simplelog.Errorf("trying to list containers from pod %v with error %v", podname, err)
+				return
+			}
+			// Loop over each container, construct a path and log file name
+			// write the output of the kubectl logs command to a file
+			for _, container := range strings.Split(containers, " ") {
+				m.Lock()
```

Comment:
so I'm I'm reading this correctly this effectively will make it single threaded anyway (at least this part of it, which is the most time consuming part). Need to figure out how to avoid the lock while still not introducing a race condition

Reply:
I removed the lock after inspecting the code and re-testing I didn't see a need for it

Reply:
so there is a race here come to find out, but it is very subtle, the createpath that is called here actually makes a directory and this is in the loop..so every thread will be competing to do that, update the code so that it just happens once outside of the loop and it should be fine